code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/AJOPayloadHandler.java

```diff
@@ -99,24 +101,40 @@ class InAppNotificationHandler {
     }
 
     /**
-     * Generates and dispatches an event prompting the Edge extension to fetch in-app messages.
+     * Generates and dispatches an event prompting the Edge extension to fetch in-app or feed messages.
      * The app surface used in the request is generated using the application id of the app.
      * If the application id is unavailable, calling this method will do nothing.
+     *
+     * @param surfacePaths A {@code List<String>} of surface path strings for fetching feed messages, if available.
      */
-    void fetchMessages() {
+    void fetchMessages(final List<String> surfacePaths) {
         final String appSurface = getAppSurface();
         if (appSurface.equals("unknown")) {
-            Log.warning(LOG_TAG, SELF_TAG, "Unable to retrieve in-app messages - unable to retrieve the application id.");
+            Log.warning(LOG_TAG, SELF_TAG, "Unable to retrieve in-app or feed messages, cannot read the bundle identifier.");
             return;
         }
 
+        final List<String> surfaceUri = new ArrayList<>();
+        if (surfacePaths != null && !surfacePaths.isEmpty()) {
+            for (final String surfacePath : surfacePaths) {
+                if (!StringUtils.isNullOrEmpty(surfacePath)) {
+                    surfaceUri.add(appSurface + File.separator + surfacePath);
```

Comment:
We should also check if the final surfaceUri makes a valid URI (probably in a future pr). I'm planning to add this validation on iOS as well.